### PR TITLE
Improvement: stick to the convention for performing on main queue.

### PR DESCRIPTION
**Pull request by krris**
Small cleaning: instead of calling `dispatch_async(dispatch_get_main_queue(), ^{})` we should use `HUBPerformOnMainQueue` just like in other places of the framework.

### CODE REVIEW

File: sources/HUBViewController.m [hunk 1]
```diff
@@ -1258,7 +1258,7 @@ - (void)loadImageFromData:(id<HUBComponentImageData>)imageData
         contextsForURL = [NSMutableArray arrayWithObject:context];
         self.componentImageLoadingContexts[imageURL] = contextsForURL;
         
-        dispatch_async(dispatch_get_main_queue(), ^{
+        HUBPerformOnMainQueue(^{
```
Comment (JohnSundell) [inline sources/HUBViewController.m]: Actually, here we want to always run the code async, to make sure that the cell that the image is for is registered with `UICollectionView` before it is loaded. We should add a test for that, that would've failed this change.

File: sources/HUBViewController.m [hunk 2]
```diff
@@ -1490,7 +1490,7 @@ - (void)scrollToRemainingComponentsOfType:(HUBComponentType)componentType
             /* This solves a case where the UICollectionView hasn't updated its visible cells until the next cycle
                when changing the content offset without animations. */
             if (!animated) {
-                dispatch_async(dispatch_get_main_queue(), stepCompletionHandler);
+                HUBPerformOnMainQueue(stepCompletionHandler);
```
Comment (JohnSundell) [inline sources/HUBViewController.m]: Same thing here. We want to do it async (the comment here says why)

**Comment by JohnSundell** [comment]
@krris Could you make this PR only contain the "valid" fix, so we can merge it in? 😄

**Comment by krris** [comment]
@JohnSundell

**Pull request merged**